kuksa_databroker/Dockerfile

```
@@ -12,31 +12,66 @@
 # ********************************************************************************/
 
 # This is expected to be executed in the kuksa.val top-level directory
-FROM rust:1.63 as builder
 
+# This builds KUKSA databroker
 
+FROM rust:1.65-alpine as base
+RUN apk add git musl-dev python3 protoc ncurses-terminfo-base make 
 RUN rustup component add rustfmt
 
+
+FROM base AS builder-amd64
+RUN echo "AMD64 quirks"
+# it seems on AMD build some dependency requires zlib headers
+# also the grpc proto compilation always falls back to
+# some binaries downloaded by cargo that require glibc
+RUN apk add zlib-dev libc6-compat  libstdc++ gcompat
+
+
+FROM base AS builder-arm64
+RUN echo "ARM64 quirks"
+# Need flag for MUSL arm builds, to prevent linker error
+# on missing __getauxval symbol, see also
+# https://github.com/rust-lang/rust/issues/89626
+ENV CFLAGS=-mno-outline-atomics
+
+
+
+FROM builder-$TARGETARCH as builder
+ARG TARGETARCH
+# This will speed up fetching the crate.io index in the future, see
+# https://blog.rust-lang.org/2022/06/22/sparse-registry-testing.html
+ENV CARGO_UNSTABLE_SPARSE_REGISTRY=true
+
+RUN echo "Building for $TARGETARCH"
+
 RUN mkdir /build
 WORKDIR /build
 
+# See also https://github.com/rust-lang/cargo/issues/10583, 
+# without this needs more than 12GB in runner if somebody wants to buildx
+# it with the common qemu setup
+RUN ln -s /usr/bin/git /usr/local/cargo/bin/
+ENV CARGO_NET_GIT_FETCH_WITH_CLI=true
+
+RUN cargo install cargo-license
+
 ADD kuksa_databroker kuksa_databroker
 ADD proto proto
```

We can do it in a later PR, but we should add the top level Cargo.toml as well since it's specifying a couple of release build options (for all members of the workspace).

This will bring the resulting binary from ~7.2MiB -> ~3.8MiB, and would possibly make it faster as well...

```suggestion
ADD proto proto
ADD Cargo.toml Cargo.toml
```